main.rb

```diff
@@ -0,0 +1,181 @@
+# frozen_string_literal: true
+
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  def my_each
+    return to_enum unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i])
+    end
+  end
+
+  def my_map(&proc)
+    temp = []
+    (0..length - 1).each do |i|
+      temp << [proc.call(self[i])] if proc
+      temp << yield(self[i]) if block_given? && !proc
+    end
+    temp = to_enum if temp.empty?
+    temp
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    temp = []
+    (0..length - 1).each do |i|
+      temp << self[i] if yield(self[i])
+    end
+    temp
+  end
+
+  def my_any?(param = true)
+    if block_given?
+      each { |item| return true if yield(item) }
+      return false
+    end
+    state = case param
+            when Regexp then check_regexp(true, param)
+            when Class then check_class(true, param)
+            else check(true, param)
+            end
+    state == true ? state : false
+  end
+
+  def my_all?(param = true)
+    if block_given?
+      each { |item| return false unless yield(item) }
+      return true
+    end
+    state = case param
+            when Regexp then check_regexp(false, param, true)
+            when Class then check_class(false, param, true)
+            else check(false, param, true)
+            end
+    state == false ? state : true
+  end
+
+  def my_none?(param = true)
+    if block_given?
+      each { |item| return false if yield(item) }
+      return true
+    end
+    state = case param
+            when Regexp then check_regexp(false, param)
+            when Class then check_class(false, param)
+            else check(false, param)
+            end
+    state == false ? state : true
+  end
```

Comment:
* [ ] my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern